device/tt_device.h

```diff
@@ -353,16 +353,6 @@ class tt_device
     virtual void broadcast_write_to_cluster(const void *mem_ptr, uint32_t size_in_bytes, uint64_t address, const std::set<chip_id_t>& chips_to_exclude,  std::set<uint32_t>& rows_to_exclude,  std::set<uint32_t>& columns_to_exclude, const std::string& fallback_tlb) {
         throw std::runtime_error("---- tt_device::broadcast_write_to_cluster is not implemented\n");
     }
-    /**
```

Comment:
What's the exact motivation for removing this? This seems just like another entrypoint, which can be convenient at some places (e.g. just pass a vector instead of vector.data() + vector.size())

Reply:
tt-metal is not using the API that uses `std::vector`, but UMD is using the function inside `tt_silicon_driver.cpp`. However, whenever you use the one with vector, you can use the one with `void*`. Tests are using the variant with `std::vector`, which doesn't make a lot of sense if we really want to test the API that tt-metal uses.

Reply:
Seems like this is adding ambiguity rather than clarity, so I agree. Thanks both for explanation.